src/provider/camunda-platform/properties/ExtensionPropertiesProps.js

```diff
@@ -63,27 +65,49 @@ function removeFactory({ commandStack, element, property }) {
       return;
     }
 
+    let values = without(properties.get('values'), property);
+
     commands.push({
-      cmd: 'properties-panel.update-businessobject-list',
+      cmd: 'element.updateModdleProperties',
       context: {
-        element: element,
-        currentObject: properties,
-        propertyName: 'values',
-        objectsToRemove: [ property ]
+        element,
+        moddleElement: properties,
+        properties: {
+          values
+        }
       }
     });
 
     // remove camunda:Properties if there are no properties anymore
-    if (properties.get('values').length === 1) {
+    if (!values.length) {
+      const businessObject = getBusinessObject(element),
+            extensionElements = businessObject.get('extensionElements');
+
+      values = without(extensionElements.get('values'), properties);
+
       commands.push({
-        cmd: 'properties-panel.update-businessobject-list',
+        cmd: 'element.updateModdleProperties',
         context: {
-          element: element,
-          currentObject: getBusinessObject(element).get('extensionElements'),
-          propertyName: 'values',
-          objectsToRemove: [ properties ]
+          element,
+          moddleElement: extensionElements,
+          properties: {
+            values
+          }
         }
       });
+
+      if (!values.length) {
+        commands.push({
+          cmd: 'element.updateModdleProperties',
+          context: {
+            element,
+            moddleElement: businessObject,
+            properties: {
+              extensionElements: undefined
```

Comment:
Can we *not* do that? Note that this behaviour is a reported bug: https://github.com/bpmn-io/bpmn-js-properties-panel/issues/499

Reply:
In that case what should we do? Remove all instances of _remove empty extension elements_?

Reply:
Yeah. When I remove an extension, I remove the item from extension elements values and only that. There can be a separate behaviour which could perform the cleanup.

Reply:
I've removed all instances of removing what we assume to be empty extension elements (cf. https://github.com/bpmn-io/bpmn-js-properties-panel/pull/536/commits/cbcf363fa4786b85341d1cc592f64dd8a2d6ff55).